### PR TITLE
Adds A bunch of user level endpoints

### PR DESCRIPTION
**Pull request by vigneshhari**
Adds Registration, Email-based validation, Password/username changes with email validation and a bunch of other stuff!

### CODE REVIEW

File: config/settings/production.py [hunk 1]
```diff
@@ -96,14 +96,11 @@
 # https://docs.djangoproject.com/en/dev/ref/settings/#email-backend
 # https://anymail.readthedocs.io/en/stable/installation/#anymail-settings-reference
 # https://anymail.readthedocs.io/en/stable/esps/mailgun/
-EMAIL_BACKEND = "anymail.backends.mailgun.EmailBackend"
+EMAIL_BACKEND = "anymail.backends.sendgrid.EmailBackend"
 ANYMAIL = {
-    "MAILGUN_API_KEY": env("MAILGUN_API_KEY", default="replace_me"),
-    "MAILGUN_SENDER_DOMAIN": env("MAILGUN_DOMAIN", default="replace_me"),
-    "MAILGUN_API_URL": env("MAILGUN_API_URL", default="https://api.mailgun.net/v3"),
+    "SENDGRID_API_KEY": env("SENDGRID_API_KEY", default="replace_me"),
```
Comment (wley3337) [inline config/settings/production.py:101]: Do we have a place we're keeping secretes?

**Review by wley3337 — APPROVED** [review]

File: config/settings/base.py [hunk 2]
```diff
@@ -331,12 +309,48 @@
...
-    "SERVE_PERMISSIONS": ["rest_framework.permissions.IsAdminUser"],
+    "SERVE_PERMISSIONS": ["rest_framework.permissions.AllowAny"],
     "SERVERS": [
         {"url": "https://127.0.0.1:8000", "description": "Local Development server"},
         {"url": "https://villagebookbuilders.org", "description": "Production server"},
         {"url": "http://127.0.0.1:8000", "description": "Local Development server"},
     ],
 }
+
+FRONTEND_URL = env("FRONTEND_URL", default="http://localhost:3000")
```
Comment (wley3337) [inline config/settings/base.py:320]: we want this to default to the Caddy setup which is: `﻿localhost.vbb.org`

File: config/settings/production.py (hunk 1 above)
Comment (vigneshhari) [reply config/settings/production.py:101]: Not at the moment, @edringger if you can get the onepassword ( or anything similar ) account setup then we can share them there.

Comment (edringger) [reply config/settings/production.py:101]: 👍 i'm working with aleem and tyler to get that set up. apparently they "tried already" but didn't find one within VBB's budget. Idk how exhaustive their search was. @AleemKhan12

**Pull request merged**